frontends/web/src/routes/account/send/components/inputs/receiver-address-input.tsx

```diff
@@ -0,0 +1,78 @@
+import { SyntheticEvent, useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { debug } from '../../../../../utils/env';
+import { getReceiveAddressList } from '../../../../../api/account';
+import DarkModeContext from '../../../../../contexts/DarkmodeContext';
+import { Input } from '../../../../../components/forms';
+import qrcodeIconDark from '../../../../../assets/icons/qrcode-dark.png';
+import qrcodeIconLight from '../../../../../assets/icons/qrcode-light.png';
+import style from '../../send.module.css';
+
+type TToggleScanQRButtonProps = {
+    onClick: () => void;
+}
+
+type TReceiverAddressInputProps = {
+    accountCode?: string;
+    addressError?: string;
+    hasCamera: boolean;
```

Comment:
could the hasCamera related code be moved into this component?

Reply:
I thought about it, but `hasCamera` depends on the `videoInputDevices.length` which comes from `qrCodeReader.getVideoInputDevices()`. The `this.qrCodeReader` is used in another component (at least for `<ScanQRDialog />`. Idk if that's possible

But, how would you do it?

Reply:
Ofc we can pass `qrCodeReader.getVideoInputDevices()` result into a prop of `<ReceiverAddressInput />` directly and evaluate the result inside the `<ReceiverAddressInput />` instead (wheter `hasCamera` or not). Is this what you're thinking?

Reply:
sorry I didn't look closely, but it feels like ScanQRDialog could all be move  into this component.